README.md
# portfolio
Link to site: https://cse134b-hw5-beab9.web.app/ 


cse134B-hw5 /changelog.md
I got pretty good scores on my lighthouse reports. Therefore I worked on making my site more mobile friendly.

cse134B-hw5 /blog.js
import { initializeApp } from "https://www.gstatic.com/firebasejs/9.6.8/firebase-app.js"
import {collection, addDoc} from 'https://www.gstatic.com/firebasejs/9.6.8/firebase-firestore.js'

        const firebaseConfig = {
        apiKey: "AIzaSyCR2HSfyjxqMwO8KDDA1Bk1KjbEMD2_nZY",
        authDomain: "cse134b-hw5-beab9.firebaseapp.com",
        projectId: "cse134b-hw5-beab9",
        storageBucket: "cse134b-hw5-beab9.appspot.com",
        messagingSenderId: "181886878266",
        appId: "1:181886878266:web:1c733c49f73dff159beb03",
        measurementId: "G-7Y1Q3L9DP1"
        };


        // Initialize Firebase
        const app = initializeApp(firebaseConfig);

        var db = firebase.firestore();
var index;
if(localStorage.getItem("blogs")==null){
    var blogs = [];
    for(let i = 0; i<5; i++){
    let blogPost = {title: 'Some Title' + i, 
                date: '05/04/2020', 
                summary: 'this is a summary'};
    blogs.push(blogPost);
}
}
else{
    blogs = JSON.parse(localStorage.getItem("blogs"));
}

function addBlog(){

    var title = document.getElementById("inputTitle");
    var date = document.getElementById("inputDate");
    var summary = document.getElementById("inputSummary");
    alert("adding");
    try {
        const docRef = await addDoc(collection(db, "blogs"), {
            title: title.value,
            date: date.value,
            summary: summary.value
        });
        console.log("Document written with ID: ", docRef.id);
      } catch (e) {
        console.error("Error adding document: ", e);
      }
    // db.collection("blogs").add({
    //     title: title.value,
    //     date: date.value,
    //     summary: summary.value
    // }).then((docRef) => {
    //     console.log("Document written with ID: ", docRef.id);
    // })
    // .catch((error) => {
    //     console.error("Error adding document: ", error);
    // });
    let blogPost = {title: title.value, date: date.value, summary: summary.value};
    //let blogPost = {title: "title", date: "date", summary: "summary"};
    
    blogs.push(blogPost);
    localStorage.setItem("blogs", JSON.stringify(blogs));
    title ='';
    date = '';
    summary = '';
}

readAll();

const addBtn = document.getElementById("addBtn");
const addDialog = document.getElementById("addDialog");
const cancelBtn = document.getElementById("cancelBtn");


addBtn.addEventListener("click", function onOpen(){
    if( typeof addDialog.showModal === "function"){
        addDialog.showModal();

    }
    else {
        alert("The <dialog> API is not supported by this browser");
    }
});


 //const form = document.querySelector("form");
 //form.addEventListener("submit", addBlog());

 const saveBtn = document.getElementById("saveBtn");
 saveBtn.addEventListener("click", addBlog);

 for(let i = 0; i< blogs.length; i++){
     const delBtn =  document.getElementById("delete" + i);
     delBtn.addEventListener("click", ()=>{
        alert("deleting");
        console.log(blogs[i]);
        deleteBlog(blogs[i]);
     })
 }

 for(let i = 0; i< blogs.length; i++){
    const updateBtn =  document.getElementById("update" + i);
    updateBtn.addEventListener("click", ()=>{
       alert("updating");
       //console.log(blogs[i]);
       editBlog(blogs[i]);

    })
}

    function readAll(){
        if(blogs.length >0){
            for(let i = 0; i<blogs.length; i++){
                        let el = document.createElement('p');
                        el.setAttribute("id", "elem" + i);
                        let deleteBtn = document.createElement('button');
                        deleteBtn.setAttribute("id", "delete" + i);
                        deleteBtn.setAttribute("class", "crudBtns");
                        let updateBtn = document.createElement('button');
                        updateBtn.setAttribute("id", "update" + i);
                        updateBtn.setAttribute("class", "crudBtns");
                        let deleteText = document.createTextNode('Delete');
                        let updateText = document.createTextNode('Update');
                        let text = document.createTextNode("Title: " + blogs[i].title + " - Date: " + blogs[i].date + " - Summary: " + blogs[i].summary );
                        el.appendChild(text);
                    
                        deleteBtn.appendChild(deleteText);
                        el.appendChild(deleteBtn);
                    
                        updateBtn.appendChild(updateText);
                        el.appendChild(updateBtn); 
                    
                        document.getElementById("blogs").appendChild(el);
                
            }
             
        }

    }

    function deleteBlog(blog){
        for(let i = 0; i<blogs.length; i++){
            if(blogs[i] != null){
                if(blogs[i].title == blog.title && blogs[i].date == blog.date && blogs[i].summary == blog.summary){
                    const row = document.getElementById("elem" + i);
                    row.remove();
                    blogs.splice(i,1);
                }
            }
        }
       localStorage.setItem("blogs", JSON.stringify(blogs));
        //console.log(blogs);
    }

    const updateDialog = document.getElementById("updateDialog");
    const cancelBtn1 = document.getElementById("cancelBtn");
    const saveUpdateBtn = document.getElementById("updateBtn");

    saveUpdateBtn.addEventListener("click", ()=>{
        updateBlogArray(index);
    });

    function editBlog(blog){
        updateDialog.showModal();
        
        for(let i =0; i<blogs.length; i++){
            if(blogs[i].title == blog.title && blogs[i].date == blog.date && blogs[i].summary == blog.summary){
                index = i;
            }
        }
        console.log(index);
        var title = document.getElementById("inputTitle1");
        title.value = blog.title;
        var date = document.getElementById("inputDate1");
        date.value = blog.date;
        var summary = document.getElementById("inputSummary1");
        summary.value = blog.summary;
        
    }

    function updateBlogArray(index){
        var title = document.getElementById("inputTitle1");
        var date = document.getElementById("inputDate1");
        var summary = document.getElementById("inputSummary1");
        if(blogs[index].title != title.value){
            console.log(blogs[index].title);
            console.log(title.value);
            blogs[index].title = title.value;
        }
        if(blogs[index].date != date.value){
            blogs[index].date = date.value;
        }
        if(blogs[index].summary != summary.value){
            blogs[index].summary = summary.value;
        }
        localStorage.setItem("blogs", JSON.stringify(blogs));
        //console.log(blogs);
    }
    //console.log(blogs)
  




cse134B-hw5 /login.js
import { initializeApp } from "https://www.gstatic.com/firebasejs/9.6.8/firebase-app.js";
        import { getAuth } from "https://www.gstatic.com/firebasejs/9.6.8/firebase-auth.js";
        import {
        connectAuthEmulator,
        signInWithEmailAndPassword,
        onAuthStateChanged, 
        signOut
        } from "https://www.gstatic.com/firebasejs/9.6.8/firebase-auth.js"
       
        const firebaseConfig = {
        apiKey: "AIzaSyCR2HSfyjxqMwO8KDDA1Bk1KjbEMD2_nZY",
        authDomain: "cse134b-hw5-beab9.firebaseapp.com",
        projectId: "cse134b-hw5-beab9",
        storageBucket: "cse134b-hw5-beab9.appspot.com",
        messagingSenderId: "181886878266",
        appId: "1:181886878266:web:1c733c49f73dff159beb03",
        measurementId: "G-7Y1Q3L9DP1"
        };


        // Initialize Firebase
        const app = initializeApp(firebaseConfig);


        // Initialize Firebase Authentication and get a reference to the service
        const auth = getAuth(app);

        //const loginForm = document.getElementById("loginForm");
        var form = document.getElementById("loginForm");
        function handleForm(event) { event.preventDefault(); } 
        form.addEventListener('submit', handleForm);
        const loginDialog = document.getElementById("loginDialog");
    


        //connectAuthEmulator(auth, "http://localhost:9099");
        const loginEmailPassword = async() =>{
        alert("checking")
        const email = document.getElementById("emailInput");
        const loginEmail = email.value;
        console.log(loginEmail);

        const password = document.getElementById("pwInput");
        const loginPassword = password.value;
            try{
                const userCred = await signInWithEmailAndPassword(auth, loginEmail, loginPassword);
                alert("user logged in");
            }catch(error){
                alert(error);
            }
        loginDialog.close()

        console.log(userCred.user);
        }

        var loginBtn2 = document.getElementById("loginBtn2");
        loginBtn2.addEventListener("click", loginEmailPassword);

        var logoutBtn = document.getElementById("logoutBtn");
        logoutBtn.addEventListener("click", () =>{
            signOut(auth);
            alert("Signed Out");
        })

        if(localStorage.getItem("blogs")==null){
            var blogs = [];
            for(let i = 0; i<5; i++){
            let blogPost = {title: 'Some Title' + i, 
                        date: '05/04/2020', 
                        summary: 'this is a summary'};
            blogs.push(blogPost);
        }
        }
        else{
            blogs = JSON.parse(localStorage.getItem("blogs"));
        }   

        onAuthStateChanged(auth, (user) => {
        if (user) {
            // User is signed in, see docs for a list of available properties
            // https://firebase.google.com/docs/reference/js/firebase.User
            let addBlogBtn = document.getElementById("addBtn");
            addBlogBtn.style.visibility = "visible";
            let welcomePrompt = document.getElementById("loggedIn");
            welcomePrompt.style.visibility = "hidden";
            welcomePrompt.innerHTML = "Logged in as " + user.email;
            for(let i = 0; i<4; i++){
                // let deleteBtn = document.getElementById("delete" + i);
                // deleteBtn.style.visibility = "visible";
                // let updateBtn = document.getElementById("update" + i);
                // updateBtn.style.visibility = "visible";               
            }
            let loginBtn = document.getElementById("loginBtn");
            loginBtn.style.visibility = "hidden";
            let logoutBtn = document.getElementById("logoutBtn");
            logoutBtn.style.visibility = "visible";
            // ...
        } else {
            // User is signed out
            // ...
            let addBlogBtn = document.getElementById("addBtn");
            addBlogBtn.style.visibility = "hidden";
            for(let i = 0; i<4; i++){
                // let deleteBtn = document.getElementById("delete" + i);
                // console.log("hi")
                // deleteBtn.style.visibility = "hidden";
                // let updateBtn = document.getElementById("update" + i);
                // updateBtn.style.visibility = "hidden";
            }
            let loginBtn = document.getElementById("loginBtn");
            loginBtn.style.visibility = "visible";
            let logoutBtn = document.getElementById("logoutBtn");
            logoutBtn.style.visibility = "hidden";
            let welcomePrompt = document.getElementById("loggedIn");
            welcomePrompt.style.visibility = "hidden";

        }
        });



cse134B-hw5 /method.js
(function(){
    var xhr;
    var date = document.getElementById("date");
    var fullDate = new Date();
    date.value = fullDate.getMonth()+1 + "/" + fullDate.getDate() + "/" + fullDate.getFullYear();
    
    document.getElementById("postBtn").addEventListener('click', makePostRequest);
    function makePostRequest(){
        var formElem = document.querySelector("form");
        var formData = new FormData(formElem);
        xhr = new XMLHttpRequest();
        var message;
        if(!xhr){
            alert("Something went wrong");
            return false;
        }
        xhr.open('POST', 'https://httpbin.org/post');
        xhr.onload = (res) =>{
            message = res['target']['response'];
            var responseOutput = document.getElementById("response");
            responseOutput.innerHTML = message;
        }
        xhr.send(formData);

        // function handleResponse(xhr){
        //     var parsedResponse = xhr.responseXML;
        //     console.log(parsedResponse);
        //     var responseOutput = document.getElementById("response");
        //     responseOutput.innerHTML = parsedResponse;
        // }
        

    }
    document.getElementById("getBtn").addEventListener('click', makeGetRequest);
    function makeGetRequest(){
        var formElem = document.querySelector("form");
        var formData = new FormData(formElem);
        xhr = new XMLHttpRequest();
        var message;
        if(!xhr){
            alert("Something went wrong");
            return false;
        }
        xhr.open('GET', 'https://httpbin.org/get');
        xhr.onload = (res) =>{
            message = res['target']['response'];
            var responseOutput = document.getElementById("response");
            responseOutput.innerHTML = message;
        }
        xhr.send(formData);
        
        // function handleResponse(xhr){
        //     var parsedResponse = xhr.responseXML;
        //     //var msg = parsedResponse.getElementsByTagName('message')[0].firstChild.nodeValue;
        //     console.log("hi");
        //     var responseOutput = document.getElementById("response");
        //     responseOutput.innerHTML = parsedResponse;
        // }
    }

    document.getElementById("putBtn").addEventListener('click', makePutRequest);
    function makePutRequest(){
        var formElem = document.querySelector("form");
        var formData = new FormData(formElem);
        xhr = new XMLHttpRequest();
        var message;
        if(!xhr){
            alert("Something went wrong");
            return false;
        }
        xhr.open('PUT', 'https://httpbin.org/put');
        xhr.onload = (res) =>{
            message = res['target']['response'];
            var responseOutput = document.getElementById("response");
            responseOutput.innerHTML = message;
        }
        xhr.send(formData);
    }

    document.getElementById("deleteBtn").addEventListener('click', makeDeleteRequest);
    function makeDeleteRequest(){
        var formElem = document.querySelector("form");
        var formData = new FormData(formElem);
        xhr = new XMLHttpRequest();
        var message;
        if(!xhr){
            alert("Something went wrong");
            return false;
        }
        xhr.open('DELETE', 'https://httpbin.org/delete');
        xhr.onload = (res) =>{
            message = res['target']['response'];
            var responseOutput = document.getElementById("response");
            responseOutput.innerHTML = message;
        }
        xhr.send(formData);
    }
})();

cse134B-hw5 /customdialog.js
const alertBtn = document.getElementById("alert");
const alertDialog = document.getElementById("alertDialog");
const okBtn = document.getElementById("okBtn");

const confirmBtn = document.getElementById("confirm");
const confirmDialog = document.getElementById("confirmDialog");
const cancelBtn = document.getElementById("cancelBtn");
const confirmBtn2 = document.getElementById("confirmBtn");
const output = document.getElementById("result");

const promptBtn = document.getElementById("prompt");
const promptDialog = document.getElementById("promptDialog");
const cancelBtn2 = document.getElementById("cancelBtn2");
const okBtn2 = document.getElementById("okBtn2");

alertBtn.addEventListener("click", function onOpen() {
    if( typeof alertDialog.showModal === "function"){
        alertDialog.showModal();

    }
    else {
        alert("The <dialog> API is not supported by this browser");
    }
});

okBtn.addEventListener("click", () =>{
    alertDialog.close();
})

confirmBtn.addEventListener("click", function onOpen() {
    if( typeof confirmDialog.showModal === "function"){
        confirmDialog.showModal();
        
    }
    else {
        alert("The <dialog> API is not supported by this browser");
    }
});

cancelBtn.addEventListener("click", () =>{
    output.innerHTML = "Confirm Result: False";
    confirmDialog.close();
})

confirmBtn2.addEventListener("click", () =>{
    output.innerHTML = "Confirm Result: True";
    confirmDialog.close();
})



promptBtn.addEventListener("click", function onOpen() {
    if( typeof promptDialog.showModal === "function"){
        promptDialog.showModal();
    }
    else {
        alert("The <dialog> API is not supported by this browser");
    }
});

cancelBtn2.addEventListener("click", () =>{
    let text = document.getElementById("input");
    output.innerHTML = "Prompt Result: The user hit cancel";
    promptDialog.close();
})

okBtn2.addEventListener("click", () =>{
    let text = document.getElementById("input");
    if(text.value == null || text.value.length==0){
        text = "The User did not enter anything";
        output.innerHTML = "Prompt Result: " + text;
    }
    else{
        let clean = DOMPurify.sanitize(text.value);
        output.innerHTML = "Prompt Result: " + clean;
    }

    promptDialog.close();
})



cse134B-hw5 /styledposts.js
import { initializeApp } from "https://www.gstatic.com/firebasejs/9.6.8/firebase-app.js"
import { getDatabase, ref, set, get, child, push, update, remove} from 'https://www.gstatic.com/firebasejs/9.6.8/firebase-database.js'
import { getAuth } from "https://www.gstatic.com/firebasejs/9.6.8/firebase-auth.js";
import {
connectAuthEmulator,
signInWithEmailAndPassword,
onAuthStateChanged, 
signOut
} from "https://www.gstatic.com/firebasejs/9.6.8/firebase-auth.js"


        const firebaseConfig = {
        apiKey: "AIzaSyCR2HSfyjxqMwO8KDDA1Bk1KjbEMD2_nZY",
        authDomain: "cse134b-hw5-beab9.firebaseapp.com",
        projectId: "cse134b-hw5-beab9",
        storageBucket: "cse134b-hw5-beab9.appspot.com",
        messagingSenderId: "181886878266",
        appId: "1:181886878266:web:1c733c49f73dff159beb03",
        measurementId: "G-7Y1Q3L9DP1"
        };


        // Initialize Firebase
        const app = initializeApp(firebaseConfig);
        const database = getDatabase();
        


        const loginDialog = document.getElementById("loginDialog");
        var loginBtn = document.getElementById("loginBtn");
        var date = document.getElementById("inputDate");
        var date1 = document.getElementById("inputDate1");
        var fullDate = new Date();
        date.value = fullDate.getMonth()+1 + "/" + fullDate.getDate() + "/" + fullDate.getFullYear();
        date1.value = fullDate.getMonth()+1 + "/" + fullDate.getDate() + "/" + fullDate.getFullYear();
        var cancel2 = document.getElementById("cancelBtn2");
        cancel2.addEventListener("click",()=>{
            loginDialog.close();
        });
window.addEventListener('DOMContentLoaded', init);
function init(){
    readAll();
}
loginBtn.addEventListener("click", function onOpen(){
    if( typeof loginDialog.showModal === "function"){
        loginDialog.showModal();
    }
    else {
        alert("The <dialog> API is not supported by this browser");
    }
});


function addBlog(){

    var title = document.getElementById("inputTitle");
    var date = document.getElementById("inputDate");
    var summary = document.getElementById("inputSummary");
    var author = document.getElementById("inputAuthor");

    if(title.value.length!=0 && date.value.length != 0 && summary.value.length != 0){
    alert("adding");
    const newPostKey = push(child(ref(database), 'blogs')).key;
    set(ref(database, 'blogs/' + title.value), {
        title: title.value,
        summary: summary.value,
        author: author.value,
        date: date.value
    });

    }
    else{
        alert("You must fill out all fields")
    }
    title ='';
    date = '';
    summary = '';
    document.location.reload(true);
}



const addBtn = document.getElementById("addBtn");
const addDialog = document.getElementById("addDialog");
const cancelBtn = document.getElementById("cancelBtn");


addBtn.addEventListener("click", function onOpen(){
    if( typeof addDialog.showModal === "function"){
        addDialog.showModal();

    }
    else {
        alert("The <dialog> API is not supported by this browser");
    }
});


 //const form = document.querySelector("form");
 //form.addEventListener("submit", addBlog());

 const saveBtn = document.getElementById("saveBtn");
 saveBtn.addEventListener("click", (e)=>{
    e.preventDefault();
    addBlog();
    addDialog.close();
 });

var count = 0;
    function readAll(){
        
        const dbRef = ref(getDatabase());
        get(child(dbRef, `blogs/`)).then((snapshot) => {
        if (snapshot.exists()) {
            snapshot.forEach(
                function(ChildSnapshot){
                    let title = ChildSnapshot.val().title;
                    let date = ChildSnapshot.val().date;
                    let author = ChildSnapshot.val().author;
                    let summary = ChildSnapshot.val().summary;
                    console.log(title);
                    addItems(title, summary, author,date, count);
                    count++;
                }
            );
        } else {
         console.log("No data available");
        }
        }).catch((error) => {
        console.error(error);
    });


    }

    function addItems(title, summary, author, date, count){
        let el = document.createElement('p');
        el.setAttribute("class", "blogs")
        let ul = document.createElement('ul');
        let liTitle = document.createElement('h2')
        let liSum = document.createElement('li');
        let liAuth = document.createElement('li');
        let liDate = document.createElement('li');
        let br = document.createElement('br');
        el.setAttribute("id", title);
        let deleteBtn = document.createElement('button');
        deleteBtn.setAttribute("id", "delete" + count);
        deleteBtn.setAttribute("class", "delete");
        let trashIcon = document.createElement("img");
        trashIcon.setAttribute("src", "./trashIcon.png");
        trashIcon.setAttribute("width", '10');

        let updateBtn = document.createElement('button');
        updateBtn.setAttribute("id", "update" + count);
        let deleteText = document.createTextNode('Delete');
        let updateText = document.createTextNode('Update');
        let updateIcon = document.createElement('img');
        updateIcon.setAttribute("src", "./editIcon.png");
        updateIcon.setAttribute("width", '10');
        let titleText = document.createTextNode("Title: " + title);
        let sumText = document.createTextNode("Content: " + summary);
        let authText = document.createTextNode("Author: " + author);
        let dateText = document.createTextNode("Date: " + date);
        liTitle.appendChild(titleText);
        liSum.appendChild(sumText);
        liAuth.appendChild(authText);
        liDate.appendChild(dateText);
        el.appendChild(ul);
        ul.appendChild(liTitle);
        ul.appendChild(liSum);
        ul.appendChild(liAuth);
        ul.appendChild(liDate);
        deleteBtn.appendChild(deleteText);
        deleteBtn.appendChild(trashIcon);
        el.appendChild(deleteBtn);

        updateBtn.appendChild(updateText);
        updateBtn.appendChild(updateIcon);
        el.appendChild(updateBtn); 

        document.getElementById("blogs").appendChild(el);

        const delBtn =  document.getElementById("delete" + count);
        delBtn.addEventListener("click", ()=>{
        //alert("deleting");
        
        deleteBlog(title);
     })

     
        const upBtn =  document.getElementById("update" + count);
        updateBtn.addEventListener("click", ()=>{
           //alert("updating");
           //console.log(blogs[i]);
           editBlog(title, date, summary, author);
    
        })
    
    }
    function deleteBlog(blog){
        console.log(blog);
        set(ref(database, 'blogs/' + blog), null);
        document.location.reload(true);
  
    }

    const updateDialog = document.getElementById("updateDialog");
    const cancelBtn1 = document.getElementById("cancelBtn");
    const saveUpdateBtn = document.getElementById("saveupdateBtn");

    saveUpdateBtn.addEventListener("click", (e)=>{
        e.preventDefault();

        var title = document.getElementById("inputTitle1");
        var date = document.getElementById("inputDate1");
        var summary = document.getElementById("inputSummary1");
        var author = document.getElementById("inputAuthor1");
        const newData = {
            title: title.value,
            summary: summary.value,
            author: author.value,
            date: date.value
        }
        console.log(date);
        const newPostKey = push(child(ref(database), 'blogs')).key;
        const updates = {};
        updates['/blogs/' + title.value] = newData;
   
        update(ref(database), updates);
        updateDialog.close();
        document.location.reload(true);
        //updateBlogArray(index);
    });

    function editBlog(title1, date1, summary1, author1){
        updateDialog.showModal();
        
        var title = document.getElementById("inputTitle1");
        title.value = title1;
        var date = document.getElementById("inputDate1");
        date.value = date1;
        var author = document.getElementById("inputAuthor1");
        author.value = author1;
        var summary = document.getElementById("inputSummary1");
        summary.value = summary1;
        
    }

    const auth = getAuth(app);

    //const loginForm = document.getElementById("loginForm");
    var form = document.getElementById("loginForm");
    function handleForm(event) { event.preventDefault(); } 
    form.addEventListener('submit', handleForm);
   



    //connectAuthEmulator(auth, "http://localhost:9099");
    const loginEmailPassword = async() =>{
    alert("checking")
    const email = document.getElementById("emailInput");
    const loginEmail = email.value;
    console.log(loginEmail);


    const password = document.getElementById("pwInput");
    const loginPassword = password.value;
        try{
            const userCred = await signInWithEmailAndPassword(auth, loginEmail, loginPassword);
            alert("user logged in");
        }catch(error){
            alert(error);
        }
        loginDialog.close()

        //console.log(userCred.user);
    }

    var loginBtn2 = document.getElementById("loginBtn2");
    loginBtn2.addEventListener("click", loginEmailPassword);

    var logoutBtn = document.getElementById("logoutBtn");
    logoutBtn.addEventListener("click", () =>{
        signOut(auth);
        alert("Signed Out");
    })

    onAuthStateChanged(auth, (user) => {
    if (user) {
        // User is signed in, see docs for a list of available properties
        // https://firebase.google.com/docs/reference/js/firebase.User
        var author = document.getElementById("inputAuthor");
        author.value = user.email;
        var author1 = document.getElementById("inputAuthor1");
        author1.value = user.email;
        let addBlogBtn = document.getElementById("addBtn");
        addBlogBtn.style.visibility = "visible";
        let welcomePrompt = document.getElementById("loggedIn");
        welcomePrompt.style.visibility = "visible";
        welcomePrompt.innerHTML = "Logged in as " + user.email;
        for(let i = 0; i<count; i++){
            let deleteBtn = document.getElementById("delete" + i);
            deleteBtn.style.visibility = "visible";
            let updateBtn = document.getElementById("update" + i);
            updateBtn.style.visibility = "visible";               
        }
        let loginBtn = document.getElementById("loginBtn");
        loginBtn.style.visibility = "hidden";
        let logoutBtn = document.getElementById("logoutBtn");
        logoutBtn.style.visibility = "visible";
        // ...
    } else {
        // User is signed out
        // ...
        let addBlogBtn = document.getElementById("addBtn");
        addBlogBtn.style.visibility = "hidden";
        for(let i = 0; i<count; i++){
            let deleteBtn = document.getElementById("delete" + i);
            console.log("hi")
            deleteBtn.style.visibility = "hidden";
            let updateBtn = document.getElementById("update" + i);
            updateBtn.style.visibility = "hidden";
        }
        let loginBtn = document.getElementById("loginBtn");
        loginBtn.style.visibility = "visible";
        let logoutBtn = document.getElementById("logoutBtn");
        logoutBtn.style.visibility = "hidden";
        let welcomePrompt = document.getElementById("loggedIn");
        welcomePrompt.style.visibility = "hidden";

    }
    });